web/app/view/ReportController.js

```diff
@@ -170,6 +184,41 @@ Ext.define('Traccar.view.ReportController', {
         });
     },
 
+    selectEvent: function (event) {
+        var position;
+        if (event.get('positionId')) {
+            position = Ext.getStore('ReportRoute').getById(event.get('positionId'));
+            if (position) {
+                this.fireEvent('selectreport', position, true);
+            }
+        }
+    },
+
+    loadEvents: function (store, data) {
+        var i, eventObject, positionIds = {};
+        Ext.getStore('ReportRoute').removeAll();
+        if (data.length > 0) {
+            for (i = 0; i < data.length; i++) {
+                eventObject = data[i];
+                if (eventObject.get('positionId')) {
+                    positionIds[eventObject.get('positionId')] = true;
```

Comment:
Can't we just use array instead? I don't expect there to be too many duplicates, so we can just de-dupe on the server side if really needed.